__tests__/app.test.js

```diff
@@ -5,66 +5,103 @@ const seed = require("../db/seeds/seed")
+            expect(article).toHaveProperty("article_id", expect.any(Number))
+            expect(article).toHaveProperty("title", expect.any(String))
+            expect(article).toHaveProperty("topic", expect.any(String))
+            expect(article).toHaveProperty("author", expect.any(String))
+            expect(article).toHaveProperty("body", expect.any(String))
+            expect(article).toHaveProperty("created_at", expect.any(String))
+            expect(article).toHaveProperty("votes", expect.any(Number))
+            expect(article).toHaveProperty("article_img_url", expect.any(String))
+            expect(article).toHaveProperty("comment_count", expect.any(String))
+          })
+        })
+    });
+    test('should return array of objects sorted by created_at property', () => {
+      return request(app).get("/api/articles")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
         });
-        test('should return array of objects sorted by created_at property', () => {
-            return request(app).get("/api/articles")
-                .expect(200)
-                .then(({ body: { articles } }) => {
-                    expect(articles).toBeSortedBy("created_at", {
-                        descending: true,
-                    });
-                });
+    });
+  });
+  describe('GET /api/articles/:article_id', () => {
+    test('should return 400 error if incorrect data type used as the parametric endpoint', () => {
+      return request(app).get("/api/articles/abc")
+        .expect(400)
+        .then(({ body: { message } }) => {
+          expect(message).toBe("Bad Request");
         });
     });
+    test('should return 404 if id not found', () => {
+      return request(app).get("/api/articles/100")
+        .expect(404)
+        .then(({ body: { message } }) => {
+          expect(message).toBe("Article Not Found")
+        })
+    });
+    test('should respond with status 200 and an object containing properties article_id, title, topic, author, body, created_at, votes and article_img_url', () => {
+      return request(app).get("/api/articles/1")
+        .expect(200)
+        .then(({ body, body: { article } }) => {
+          console.log(body);
+          expect(body).toHaveProperty("article")
+          expect(article).toHaveProperty("article_id", expect.any(Number))
+          expect(article).toHaveProperty("title", expect.any(String))
+          expect(article).toHaveProperty("topic", expect.any(String))
+          expect(article).toHaveProperty("author", expect.any(String))
+          expect(article).toHaveProperty("body", expect.any(String))
+          expect(article).toHaveProperty("created_at", expect.any(String))
+          expect(article).toHaveProperty("votes", expect.any(Number))
+          expect(article).toHaveProperty("article_img_url", expect.any(String))
+        })
```

Comment:
I think this could do with cleaning up as a test - have you had a look at more flexible matches such as expect.objectContaining or .toMatchObject?

Reply:
Good idea. Thanks, I have refactored using .toMatchObject.